packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/private/helpers.ts

```diff
@@ -91,3 +91,39 @@ export function createEncryptionConfig(role: iam.IRole, encryptionKey?: kms.IKey
     ? { kmsEncryptionConfig: { awskmsKeyArn: encryptionKey.keyArn } }
     : { noEncryptionConfig: 'NoEncryption' };
 }
+
+export function createProcessingConfig(
+  scope: Construct,
+  role: iam.IRole,
+  dataProcessors?: firehose.IDataProcessor[],
+): firehose.CfnDeliveryStream.ProcessingConfigurationProperty | undefined {
+  if (dataProcessors && dataProcessors.length > 1) {
+    throw new Error('Only one processor is allowed per delivery stream destination');
+  }
+  if (dataProcessors && dataProcessors.length > 0) {
+    const processors = dataProcessors.map((processor) => {
+      const processorConfig = processor.bind(scope, { role });
+      const parameters = [{ parameterName: 'RoleArn', parameterValue: role.roleArn }];
+      parameters.push(processorConfig.processorIdentifier);
+      if (processor.props.bufferInterval) {
+        parameters.push({ parameterName: 'BufferIntervalInSeconds', parameterValue: processor.props.bufferInterval.toSeconds().toString() });
+      }
+      if (processor.props.bufferSize) {
+        parameters.push({ parameterName: 'BufferSizeInMBs', parameterValue: processor.props.bufferSize.toMebibytes().toString() });
+      }
+      if (processor.props.retries) {
+        parameters.push({ parameterName: 'NumberOfRetries', parameterValue: processor.props.retries.toString() });
+      }
+      return {
+        type: processorConfig.processorType,
+        parameters: parameters,
+      };
+    });
```

Comment:
Any chance you can extract this to a separate file-private function? I think it will help readability.